flink-kubernetes/src/main/java/org/apache/flink/kubernetes/cli/FlinkKubernetesCustomCli.java

```diff
@@ -0,0 +1,294 @@
+
+	// Options with short prefix(k) or long prefix(kubernetes)
+	private final Option imageOption;
+	private final Option clusterIdOption;
+	private final Option jobManagerMemeoryOption;
+	private final Option taskManagerMemeoryOption;
+	private final Option taskManagerSlotsOption;
+	private final Option dynamicPropertiesOption;
+	private final Option helpOption;
+
+	private final Option jobClassOption;
+	private final Option jobIdOption;
+
+	private final ClusterClientServiceLoader clusterClientServiceLoader;
+
+	public FlinkKubernetesCustomCli(
+		Configuration configuration,
+		String shortPrefix,
+		String longPrefix) {
+		this(configuration, new DefaultClusterClientServiceLoader(), shortPrefix, longPrefix);
+	}
+
+	public FlinkKubernetesCustomCli(
+		Configuration configuration,
+		ClusterClientServiceLoader clusterClientServiceLoader,
+		String shortPrefix,
+		String longPrefix) {
+		super(configuration);
+
+		this.clusterClientServiceLoader = clusterClientServiceLoader;
+
+		this.imageOption = KubernetesCliOptions.getOptionWithPrefix(IMAGE_OPTION, shortPrefix, longPrefix);
+		this.clusterIdOption = KubernetesCliOptions.getOptionWithPrefix(CLUSTER_ID_OPTION, shortPrefix, longPrefix);
+		this.dynamicPropertiesOption = KubernetesCliOptions.getOptionWithPrefix(
+			DYNAMIC_PROPERTY_OPTION, shortPrefix, longPrefix);
+
+		this.jobManagerMemeoryOption = KubernetesCliOptions.getOptionWithPrefix(
+			KubernetesCliOptions.JOB_MANAGER_MEMORY_OPTION, shortPrefix, longPrefix);
+		this.taskManagerMemeoryOption = KubernetesCliOptions.getOptionWithPrefix(
+			KubernetesCliOptions.TASK_MANAGER_MEMORY_OPTION, shortPrefix, longPrefix);
+		this.taskManagerSlotsOption = KubernetesCliOptions.getOptionWithPrefix(
+			KubernetesCliOptions.TASK_MANAGER_SLOTS_OPTION, shortPrefix, longPrefix);
+
+		this.jobClassOption = KubernetesCliOptions.getOptionWithPrefix(JOB_CLASS_NAME_OPTION, shortPrefix, longPrefix);
+		this.jobIdOption = KubernetesCliOptions.getOptionWithPrefix(JOB_ID_OPTION, shortPrefix, longPrefix);
+
+		this.helpOption = KubernetesCliOptions.getOptionWithPrefix(HELP_OPTION, shortPrefix, longPrefix);
+	}
+
+	/**
+	 * active if commandline contains option -m kubernetes-cluster or -kid.
+	 */
+	@Override
+	public boolean isActive(CommandLine commandLine) {
+		final String jobManagerOption = commandLine.getOptionValue(addressOption.getOpt(), null);
+		final boolean k8sJobManager = ID.equals(jobManagerOption);
+		final boolean k8sClusterId = commandLine.hasOption(clusterIdOption.getOpt());
+		return k8sJobManager || k8sClusterId;
+	}
+
+	@Override
+	public void addRunOptions(Options baseOptions) {
+		baseOptions.addOption(DETACHED_OPTION)
+			.addOption(imageOption)
+			.addOption(clusterIdOption)
+			.addOption(jobManagerMemeoryOption)
+			.addOption(taskManagerMemeoryOption)
+			.addOption(taskManagerSlotsOption)
+			.addOption(dynamicPropertiesOption)
+			.addOption(helpOption);
+	}
+
+	@Override
+	public void addGeneralOptions(Options baseOptions) {
```

Comment:
Why don't you call `super.addGeneralOptions(baseOptions);`?

Reply:
I can see the address can be only `kubernetes-cluster` but don't we respect zookeeper namespace if user explicitly specific one?

Reply:
Because we do not need the following run options. I also think the they are unnecessary for yarn-cluster. Because they are so confusing.
```
    -m,--jobmanager <arg>                Address of the JobManager (master) to
                                          which to connect. Use this flag to
                                          connect to a different JobManager than
                                          the one specified in the
                                          configuration.
    -z,--zookeeperNamespace <arg>        Namespace to create the Zookeeper
                                          sub-paths for high availability mode
```

Reply:
OK I agree about `-m` but possibly `-z` is still valid. We can defer the decision since always setting zk namespace as cluster id in k8s case works as is.